src/com/sap/piper/Notify.groovy

```diff
@@ -0,0 +1,51 @@
+package com.sap.piper
+
+import groovy.text.SimpleTemplateEngine
+
+class Notify implements Serializable {
+    private static enum Severity { ERROR, WARNING }
+    private final static String LIBRARY_NAME = 'piper-lib-os'
+    private final static String MESSAGE_PATTERN = '[${severity}] ${message} (${libName}/${stepName})'
+
+    protected static Utils instance = null
+
+    protected static Utils getUtilsInstance(){
+        instance = instance ?: new Utils()
```

Comment:
What is the benefit with instanciating the util class when we call log the first time (this is acutally how the (only possible) code flow looks like. Why not instanciating when the variable is declared? Would we simpler and less code.

Reply:
I would like to use the `Telemetry` singleton here, I have a look if I can refactor this.